### DISCUSSION TITLE
SQL syntax and date/epoch conversion

### DISCUSSION THREAD

**Discussion by helander** [pocketbase/pocketbase#3070] [Q&A] [answered]
Is it possible in a "View" collection to convert column values from epoch to date using some SQL function? Other SQL dialects have a "to_timestamp()" function that does this.

The use case is that I often store epoch numbers in the database, but when I want to look at the data using the admin UI, I would like to display these values in some human readable format (and not a big integer number).

SELECT id, to_timestamp(time), value FROM mycollection

**Comment by ganigeorgiev** [comment, marked as answer]
SQLite also have builtin date functions. You can explore https://www.sqlite.org/lang_datefunc.html.

_The only caveat I forgot to mention when using a view expression is that the expression column must have a valid alias so that it can be used as field name, eg. `strftime('%Y-%m-%d', ...) as mydate`._

**Reply by helander** [reply]
One thing I have considered is to change the epoch number columns in my collections to Pocketbase ‘datetime’. What gets physically stored in the database for a ‘datetime’ column, is it a UTC formatted string (as you see in the UI) or is it actually stored as an integer value? Is there some tool that can update a column with epoch numbers to a column with corresponding ‘datetime’ values (or do I in that case have to write a program to perform the update) ?

**Reply by ganigeorgiev** [reply]
Collection's `date` field values are stored as plain string in the format `Y-m-d H:i:s.uZ` (_it is similar to a iso8601 format but with a space as demiliter instead of `T` for backward compatibility reasons with earlier versions; once we introduce date filters normalization this may change_).

> Is there some tool that can update a column with epoch numbers to a column with corresponding ‘datetime’ values (or do I in that case have to write a program to perform the update) ?

You can perform the update using the SQL `UPDATE` statement, either run directly against the db file with `sqlite3 pb_data/data.db "..."` or with PocketBase as migration or console command (see [Execute raw statements](https://pocketbase.io/docs/js-database/#execute)). An example query could be (**I haven't tested it**):

```sql
UPDATE yourTable
SET yourColumn = strftime('%Y-%m-%d %H:%M:%fZ', yourColumn, 'unixepoch')
```

_You can run a select just in case before applying the above to ensure that is correct, eg:_
```sql
SELECT id, strftime('%Y-%m-%d %H:%M:%fZ', yourColumn, 'unixepoch') as date
FROM yourTable
````

**Reply by helander** [reply]
Thank you very much 😀
